Makefile

```diff
@@ -488,6 +489,22 @@ ifeq ("$(with_rdpclient)", "yes")
 		cargo build -p rdp-client $(if $(FIPS),--features=fips) --release --locked $(CARGO_TARGET)
 endif
 
+.PHONY: build-ironrdp-wasm
+build-ironrdp-wasm: ironrdp = web/packages/shared/libs/ironrdp
+build-ironrdp-wasm: ensure-wasm-deps
+	cargo build --package ironrdp --lib --target $(CARGO_TARGET_wasm) --release
+	wasm-opt target/$(CARGO_TARGET_wasm)/release/ironrdp.wasm -o target/$(CARGO_TARGET_wasm)/release/ironrdp.wasm -O
+	wasm-bindgen target/$(CARGO_TARGET_wasm)/release/ironrdp.wasm --out-dir $(ironrdp)/pkg --typescript --target web
+	@echo "*" > $(ironrdp)/pkg/.gitignore
+	@printf '%s\n' '{' \
+	' "name": "ironrdp",' \
+	' "version": "0.1.0",' \
+	' "module": "ironrdp.js",' \
+	' "types": "ironrdp.d.ts",' \
+	' "files": ["ironrdp_bg.wasm","ironrdp.js","ironrdp.d.ts"],' \
+	' "sideEffects": ["./snippets/*"]' \
+	'}' > $(ironrdp)/pkg/package.json
```

Comment:
Another way to do this which may be a bit easier to modify later:
```
define ironrdp_package_json
{
  "name": "ironrdp",
  "version": "0.1.0",
   ...
}
endef
export ironrdp_package_json
```
and in the recipe:
```
    printenv ironrdp_package_json > $(ironrdp)/pkg/package.json
```

The source material is a little bit easier to edit as it does not need all those single quotes and backslashes, which are easy to get wrong. It's just bare json which is also easier to cut and paste.